### ISSUE TITLE
Getting Faraday::ResourceNotFound errors for get_deleted calls

### ISSUE THREAD

**Issue by mmmries** [restforce/restforce#385]
In my code I'm calling `client.get_deleted("Account", 7.days.ago, Time.current)` to ask for any accounts that have been recently deleted in salesforce, but this is raising a `Faraday::ResourceNotFound (NOT_FOUND: The requested resource does not exist)` error.

Was this facility removed from the salesforce API? Or Maybe moved?

**Comment by mmmries** [comment]
BTW I've been on version `2.5.3` for a while now and this was working as of last night when my system last ran an update, so maybe this is an outage on the salesforce side?

**Comment by mmmries** [comment]
I dug into this further and it appears to be a problem where the client is adding an extra `/` into the URL. If i turn on logging in the client and call `client.get_deleted` I get a log entry like this:

```
  url: "https://na34.salesforce.com/services/data/v42.0//sobjects/Opportunity/deleted/?end=2018-06-07T18%3A04%3A34Z&start=2018-06-01T18%3A04%3A34Z"
  method: :get
  headers: {"User-Agent"=>"Faraday v0.12.2", "Authorization"=>"OAuth AN-OAUTH-TOKEN"}
  body: nil

  status: "404"
  headers: {"date"=>"Fri, 08 Jun 2018 18:04:34 GMT", "strict-transport-security"=>"max-age=31536000; includeSubDomains", "x-content-type-options"=>"nosniff", "x-xss-protection"=>"1; mode=block", "content-security-policy"=>"upgrade-insecure-requests", "x-robots-tag"=>"none", "cache-control"=>"no-cache,must-revalidate,max-age=0,no-store,private", "set-cookie"=>"BrowserId=k--RO1BcQp2IxHKC4tq7eg;Path=/;Domain=.salesforce.com;Expires=Tue, 07-Aug-2018 18:04:34 GMT;Max-Age=5184000", "expires"=>"Thu, 01 Jan 1970 00:00:00 GMT", "sforce-limit-info"=>"api-usage=18232/44000", "content-type"=>"application/json;charset=UTF-8", "transfer-encoding"=>"chunked", "connection"=>"close"}
  body: "[{\"errorCode\":\"NOT_FOUND\",\"message\":\"The requested resource does not exist\"}]"
```

If I copy that same URL, but remove the extra `/` just before `sobjects`, then I can make a curl request on the comandline and it works fine.

**Comment by mmmries** [comment]
I did some additional testing and this call works fine under `ruby-2.4.3`, but under `ruby-2.5.1` I get the error above. It also looks like this was already fixed in d51befa7f74e54cece9cd553473edba07673b73f, but the gem hasn't been re-released since then?

**Comment by mmmries** [comment]
Looks like this is actually fixed by #366, I'll just wait for a release that includes that fix.
